find-closest-facility-from-point/src/main/java/com/esri/arcgismaps/sample/findclosestfacilityfrompoint/components/MapViewModel.kt

```diff
@@ -0,0 +1,214 @@
+    val map = ArcGISMap(BasemapStyle.ArcGISStreets).apply {
+        initialViewpoint = Viewpoint(
+            latitude = 32.727, longitude = -117.1750, scale = 100000.0
+        )
+    }
+
+    // Hardcoded list of facilities
+    private var facilitiesList = listOf(
+        Facility(Point(-1.3042129900625112E7, 3860127.9479775648, SpatialReference.webMercator())),
+        Facility(Point(-1.3042193400557665E7, 3862448.873041752, SpatialReference.webMercator())),
+        Facility(Point(-1.3046882875518233E7, 3862704.9896770366, SpatialReference.webMercator())),
+        Facility(Point(-1.3040539754780494E7, 3862924.5938606677, SpatialReference.webMercator())),
+        Facility(Point(-1.3042571225655518E7, 3858981.773018156, SpatialReference.webMercator())),
+        Facility(Point(-1.3039784633928463E7, 3856692.5980474586, SpatialReference.webMercator())),
+        Facility(Point(-1.3049023883956768E7, 3861993.789732541, SpatialReference.webMercator()))
+    )
+
+    // Create graphic overlay of facilities to display them on the map when user opens app
+    private val facilityGraphicsOverlay = GraphicsOverlay()
+
+    // Create graphic overlay of incident to display the incident once user taps on map
+    val incidentGraphicsOverlay = GraphicsOverlay()
+
+    val graphicsOverlays = listOf(facilityGraphicsOverlay, incidentGraphicsOverlay)
+
+
+    init {
+        createFacilitiesAndGraphics()
+    }
+
+    /**
+     * Retrieve the incident point and initiate the process to finding the route between the incident and closest facility
+     *
+     */
+    fun onSingleTapConfirmed(
+        currentJob: MutableState<Job?>,
+        event: SingleTapConfirmedEvent,
+        incidentGraphicsOverlay: GraphicsOverlay
+    ) {
+
+        // Cancel previous job
+        currentJob.value?.cancel()
+
+        // Retrieve the tapped map point from the SingleTapConfirmedEvent
+        val mapPoint: Point =
+            event.mapPoint ?: return showMessage("No map point retrieved from tap.")
+
+        val incidentPoint = Incident(mapPoint)
+
+        clearIncident(incidentGraphicsOverlay)
+        addIncident(incidentPoint, incidentGraphicsOverlay)
+        currentJob.value = viewModelScope.launch {
+            findRoute(Incident(mapPoint), incidentGraphicsOverlay)
+        }
+    }
+
+    /**
+     * Facility markers are drawn on the map right when the user opens app
+     *
+     */
+    private fun createFacilitiesAndGraphics() {
+
+        // Add all facility centers on the map
+        val facilityUrl = application.getString(R.string.hospital_symbol_url)
+        val facilitySymbol = PictureMarkerSymbol(facilityUrl)
+        facilitySymbol.height = 30F
+        facilitySymbol.width = 30F
+        for (facility in facilitiesList) {
+            val facilityGraphic = Graphic(
+                geometry = facility.geometry, symbol = facilitySymbol
+            )
+            facilityGraphicsOverlay.graphics.add(facilityGraphic)
+        }
```

Comment:
For Compose function call with multiple named arguments, let's use a new line for each argument to help with the readability:

```suggestion
        // Create a graphic for each facility and add them to the graphics overlay
        for (facility in facilitiesList) {
            val facilityGraphic = Graphic(
                geometry = facility.geometry,
                symbol = facilitySymbol
            )
            facilityGraphicsOverlay.graphics.add(facilityGraphic)
        }
```